### PR TITLE
Bug 1173097 - [Contacts][NGA] Extract 'goToSelectTag' functionality to it own .js class

### PR DESCRIPTION
**Pull request by jpruden92**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0d7c390776122ed7ca3bb151666a23899f6999b9

File: apps/communications/contacts/js/utilities/tagSelector.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      customTagReset = document.getElementById('custom-tag-reset');
+      customTagReset.addEventListener('touchstart', handleCustomTagReset);
+    }
+    if (!tagDone) {
+      tagDone = document.querySelector('#settings-done');
+      tagDone.addEventListener('click', handleSelectTagDone);
+    }
+    if (!tagHeader) {
+      tagHeader = document.querySelector('#settings-header');
+      tagHeader.addEventListener('action', Contacts.handleBack);
```
Comment (borjasalguero) [inline apps/communications/contacts/js/utilities/tagSelector.js]: Could we accept 'handleBack' as a param?

File: apps/communications/contacts/js/utilities/tagSelector.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  function handleCustomTag(ev) {
+    if (ev.keyCode === 13) {
+      ev.preventDefault();
+    }
+    ContactsTag.touchCustomTag();
+  }
+
+  function showSelectTag() {
```
Comment (borjasalguero) [inline apps/communications/contacts/js/utilities/tagSelector.js]: I would rename this as `showSelectTag(cancelCB)`, so we can remove the implicit dependency with contacts

File: apps/communications/contacts/js/utilities/tagSelector.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        }
+      });
+      contactTag.dispatchEvent(valueModifiedEvent);
+      Contacts.handleBack();
+    });
+  }
+
+  var TagSelector = {
+
+	  goToSelectTag: function goToSelectTag(event) {
```
Comment (borjasalguero) [inline apps/communications/contacts/js/utilities/tagSelector.js]: What about changing this to `show(contactTag, cancelCB)` instead of `event`?

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f5559f86616771aabe069055b61bf3860ee0ed65

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=38227327de3a1576c3e57d0563025b013c2986f5

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fbe92e3894df55157e0e4afb72901e4beee0a42e

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b2035ad67ce71cf2350711901f21205a0b331e06

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=657da1b2e85939f4e2febd0709c966ae6ade8c73

File: apps/communications/contacts/js/bootstrap.js [hunk 4]
```diff
@@ -12,6 +12,7 @@
   function loadScripts() {
     var dependencies = [
      '/contacts/services/contacts.js',
+     '/contacts/js/utilities/tagSelector.js',
```
Comment (arcturus) [inline apps/communications/contacts/js/bootstrap.js]: Don't load the tag selector on the critical path of the app, won't be needed on startup time.
A good place to lazy load this app will be at the moment that we click on the tags, in `onGoToSelectTag` seems to be a good candidate.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=67d3a6c8ea856af5c31cf3cbb77976c917af1a2b

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=62d680a841f399f9cb0a71cb337793d714e90c33

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3d5e65d6fe8154ad8e0c06011c5c52976c0ff7e6

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d778a1b03922fcd4c46ec2f095bc905d1cc71d43

**Pull request merged**